src/media/js/mkt-carousel.js

```diff
@@ -0,0 +1,150 @@
+define('mkt-carousel', ['defer'], function (defer) {
+    /*
+     * A carousel that is mostly intended for the desktop promo. Offsets are
+     * managed in CSS so the JS doesn't deal with that. The carousel only
+     * supports cycling and requires at least as many items as are visible at
+     * rest.
+     *
+     * Flexbox ordering is used to reorder the items without modifying the DOM.
+     * Cycling works by cloning the desktop-promo-items for use as
+     * placeholders. When at rest there are no placeholders visible. When the
+     * carousel is rotating one placeholder will be shown in the position of
+     * the item that is cycling around to the other side.
+     *
+     * <div class="desktop-promo">
+     *   <div class="desktop-promo-nav-left">&lt;</div>
+     *   <div class="desktop-promo-items">
+     *     <div class="desktop-promo-item">My awesome content!</div>
+     *     <!-- Insert more awesome content -->
+     *   </div>
+     *   <div class="desktop-promo-nav-right">&gt;</div>
+     * </div>
+     */
+
+    function initialize() {
+        var root = document.querySelector('.desktop-promo');
+        var previousButton = root.querySelector('.desktop-promo-nav-left');
+        var nextButton = root.querySelector('.desktop-promo-nav-right');
+        var desktopPromo = root.querySelector('.desktop-promo-items');
+        var promoItems = Array.prototype.slice.call(desktopPromo.children);
+        var placeholderItems = promoItems.map(function(item) {
+            var placeholder = item.cloneNode();
```

Comment:
Couldn't this be cloneNode(true) to get the children to save the need for the following line?